types/event.go

```diff
@@ -128,12 +128,12 @@ func (e *Event) IsResolution() bool {
 	}
 
 	// Try to retrieve the previous status in the check history and verify if it
-	// was a non-zero status, therefore indicating a resolution
-	isResolution := (len(e.Check.History) > 0 &&
-		e.Check.History[len(e.Check.History)-1].Status != 0 &&
+	// was a non-zero status, therefore indicating a resolution. The current event
+	// has already been added to the check history by eventd so we must retrieve
+	// the second to the last
+	return (len(e.Check.History) > 1 &&
+		e.Check.History[len(e.Check.History)-2].Status != 0 &&
```

Comment:
This is quite confusing. We should either:
- have the history include the current check and not have `Check.Status`
- have the history not include the current check and have `Check.Status`
But not both at the same time, as it duplicates information and can lead to inconsistencies. `TestEventIsResolution()` shows that.

Reply:
I think the first option should be avoided since it would be a major breaking change.

The second one is probably the right one but it would definitely be a breaking change, both in our code and for our users since a lot of logic (especially the flapping one) is expecting the current status in the history. Because of that, I would suggest we create a new issue so we can ship this bugfix ASAP and then we can take the time to redesign the check history!

Reply:
Just created https://github.com/sensu/sensu-go/issues/2093 so we can keep track of that!